File: src/AoC16/s16e01-typescript/common.test.ts
import { describe, test } from "node:test";
import assert from "node:assert";
import { solvePart1, solvePart2 } from "./common.js";

const sampleInput = `R2, L3`;

describe("Part 1", () => {
  test("should solve sample input correctly", () => {
    const result = solvePart1(sampleInput);
    assert.strictEqual(result, 5);
  });
});

describe("Part 2", () => {
  test("should solve sample input correctly", () => {
    const result = solvePart2("R8, R4, R4, R8");
    assert.strictEqual(result, 4);
  });
});

File: src/AoC25/s25e09-kotlin/test/kotlin/CommonTest.kt
// Simple Kotlin test runner (no JUnit dependency)

const val SAMPLE_INPUT = """
7,1
11,1
11,7
9,7
9,5
2,5
2,3
7,3
"""

fun assertEquals(
    expected: Long,
    actual: Long,
    message: String,
) {
  if (expected != actual) {
    throw AssertionError("$message: expected $expected but got $actual")
  }
}

fun runTests() {
  // Add test functions here
  test1()
  test2()

  println("✅ All tests passed!")
}

fun test1() {
  val expected = 50L
  val result = solvePart1(SAMPLE_INPUT)
  assertEquals(expected, result, "Part 1 sample test")
}

fun test2() {
  // TODO: Add sample input and expected output from problem description
  val input =
      """
        sample input here
        """
          .trimIndent()
  val expected = 0L // TODO: Update with expected value
  val result = solvePart2(input)
  assertEquals(expected, result, "Part 2 sample test")
}

fun main() {
  runTests()
}
